rust/src/minify/pass1.rs

```diff
@@ -419,7 +419,7 @@ impl<'a, 'b> Visitor<'a> for Pass1<'a, 'b> {
               .hoisted_vars
               .extend_from_slice(&alt_expr.hoisted_vars);
             // Due to normalisation, it's not possible for an `if-else` to return in either branch, because one branch would've been unwrapped.
-            assert!(cons_expr.returns && alt_expr.returns);
+            assert!(!(cons_expr.returns && alt_expr.returns));
```

Comment:
```suggestion
            assert!(cons_expr.returns == alt_expr.returns);
```

This is what I assume it was supposed to be after reading through the process_if_branch function, making sure they still both return or both don't return.

Reply:
ah yes I see.

A quick test case of confirms this
```js
if (a) {
  return false;
} else {
  return true;
}
```
This fails the assertion in my commit but works with your fix. Thanks for spotting that :)